crates/nu-command/src/network/http/client.rs

```diff
@@ -221,132 +223,171 @@ pub fn send_request(
+                return Err(ShellErrorOrRequestError::ShellError(ShellError::UnsupportedHttpBody {
+                    msg: "Body type 'List' for form requests requires paired values. E.g.: [value, 10]".into(),
+                }));
+            }
 
-                    let request_fn =
-                        move || req.set("Content-Type", &content_type).send_bytes(&data);
+            let data = vals
+                .chunks(2)
+                .map(|it| Ok((it[0].coerce_string()?, it[1].coerce_string()?)))
+                .collect::<Result<Vec<(String, String)>, ShellErrorOrRequestError>>()?;
 
-                    send_cancellable_request(&request_url, Box::new(request_fn), span, signals)
-                }
-                Value::List { vals, .. } if body_type == BodyType::Form => {
-                    if vals.len() % 2 != 0 {
-                        return Err(ShellErrorOrRequestError::ShellError(ShellError::IOError {
-                            msg: "unsupported body input".into(),
-                        }));
-                    }
+            let request_fn = Box::new(|| build_request_fn(data));
+            send_cancellable_request(request_url, request_fn, span, signals)
+        }
+        Value::Record { val, .. } => {
+            let mut data: Vec<(String, String)> = Vec::with_capacity(val.len());
 
-                    let data = vals
-                        .chunks(2)
-                        .map(|it| Ok((it[0].coerce_string()?, it[1].coerce_string()?)))
-                        .collect::<Result<Vec<(String, String)>, ShellErrorOrRequestError>>()?;
-
-                    let request_fn = move || {
-                        // coerce `data` into a shape that send_form() is happy with
-                        let data = data
-                            .iter()
-                            .map(|(a, b)| (a.as_str(), b.as_str()))
-                            .collect::<Vec<(&str, &str)>>();
-                        req.send_form(&data)
-                    };
-                    send_cancellable_request(&request_url, Box::new(request_fn), span, signals)
-                }
-                Value::List { .. } if body_type == BodyType::Json => {
-                    let data = value_to_json_value(&body)?;
-                    send_cancellable_request(
-                        &request_url,
-                        Box::new(|| req.send_json(data)),
-                        span,
-                        signals,
-                    )
+            for (col, val) in val.into_owned() {
+                data.push((col, val.coerce_into_string()?))
+            }
+
+            let request_fn = Box::new(|| build_request_fn(data));
+            send_cancellable_request(request_url, request_fn, span, signals)
+        }
+        _ => Err(ShellErrorOrRequestError::ShellError(
+            ShellError::UnsupportedHttpBody {
+                msg: format!("Accepted types: [List, Record]. Check: {HTTP_DOCS}"),
```

Comment:
(apart from the choice of error variant) by convention our types are given in lowercase as this is also required by our syntax for type annotation